### PR TITLE
Refactor to use Chef

### PR DESCRIPTION
**Pull request by andy-dufour**
Instead of shell scripts, we're bootstrapping using chef-solo to build the Chef server, and then hooking up to that Chef server to build the rest.

Needs some more ❤️  in the future, a couple of custom resources to manage delivery-ctl and add one for the key addition.

Pay attention to the read me if you test this out - it expects a couple things in the .chef directory of the project.

I have berks vendor as a local-exec in main.tf so you shouldn't need to do it as I stated in the readme but I haven't tested that thoroughly yet.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,8 +43,21 @@ This project assumes that your security team has already created VPCs, security_
...
 
+# Delivery validator keys
+
+You must provide your own Delivery validator key.
+
 # Builder keys
 
-You must provide a public/private builder key pair. They must be in ``.keys/builder_key` and ``.keys/builder_key.pub` for now.
+You no longer need to provide builder keys. Once you've generated a `delivery-validator.pem` using `ssh-keygen` or your other favorite key generator, you can generate the public key in the format needed by the Chef server by using a command like `openssl rsa -in delivery-validator.pem -pubout -out delivery-validator.pub`.
+
```
Comment (irvingpop) [inline README.md]: For now, recommend updating the instructions "for dummies" (like me) to be:

``` bash
ssh-keygen -t rsa -N "" -f .chef/delivery-validator.pem
openssl rsa -in .chef/delivery-validator.pem -pubout -out .chef/delivery-validator.pub
```

File: cookbooks/chef_server/recipes/_push_server.rb [hunk 2]
```diff
@@ -0,0 +1,4 @@
+
+chef_ingredient 'push-jobs-server' do
```
Comment (irvingpop) [inline cookbooks/chef_server/recipes/_push_server.rb:2]: I noticed this is installing opscode-push-jobs-server-2.1.0-1.el7.x86_64.rpm  -  that's bad right?

File: cookbooks/chef_server/recipes/_save_secrets.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
+#TODO: Replace with chef vault or similar. Non-encrypted databags == bad.
```
Comment (irvingpop) [inline cookbooks/chef_server/recipes/_save_secrets.rb:1]: This is super clever.  I love it

File: README.md (hunk 1 above)
Comment (andy-dufour) [reply README.md]: I just added local exec steps in the tf plan to deal with this, why make people type :)

File: run.sh [hunk 4]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+mkdir .chef
+ssh-keygen -t rsa -N "" -f .chef/delivery-validator.pem -y
+openssl rsa -in .chef/delivery-validator.pem -pubout -out .chef/delivery-validator.pub
```
Comment (irvingpop) [inline run.sh]: should this also run `terraform apply` ?  or else rename to setup.sh ?

**Pull request merged**